src/gleam/list.gleam

```diff
@@ -246,6 +246,36 @@ pub fn map(list: List(a), with fun: fn(a) -> b) -> List(b) {
   do_map(list, fun, [])
 }
 
+/// Similar to map but also lets you pass around an accumulated value.
+///
+/// ## Examples
+///
+/// ```
+/// > map_reduce(
+///     over: [1, 2, 3],
+///     from: 100,
+///     with: fn(memo, i) { tuple(memo + i, i * 2) }
+///  )
+///  tuple(106, [2, 4, 6])
+/// ```
+///
+pub fn map_reduce(
+  over list: List(a),
+  from memo: memo,
+  with fun: fn(memo, a) -> tuple(memo, b),
```

Comment:
Could we flip the arguments here? To match `fold` and `fold_right`

Reply:
Done